auditor/auditor.go

```diff
@@ -34,17 +34,17 @@ type AuditEvent struct {
 
 type Auditor struct {
 	opa             *opa.OpaAuditor
-	entitiesWatcher *entities.EntitiesWatcher
+	entitiesWatcher entities.EntitiesWatcherInterface
 	sendAuditResult agent.AuditResultHandler
 
 	auditEvents  chan AuditEvent
 	ctx          context.Context
 	cancelWorker context.CancelFunc
 }
 
-func NewAuditor(parentsStore *kuber.ParentsStore, entitiesWatcher *entities.EntitiesWatcher) *Auditor {
+func NewAuditor(entitiesWatcher entities.EntitiesWatcherInterface) *Auditor {
 	a := &Auditor{
-		opa:             opa.New(parentsStore),
+		opa:             opa.New(entitiesWatcher.ParentsStore()),
```

Comment:
The auditor is still getting the parent store from the entities watch so that doesn't change anything. The entities watch interface should hide the implementation of the parent store and all calls should be made to the entities watcher directly